homeassistant/components/smartthings/__init__.py

```diff
@@ -143,3 +143,28 @@ async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
 
     return True
+
+
+def process_status(
+    status: dict[str, dict[Capability, dict[Attribute, Status]]],
+) -> dict[str, dict[Capability, dict[Attribute, Status]]]:
+    """Remove disabled capabilities from status."""
+    if (main_component := status.get("main")) is None:
+        return status
+    if (
+        disabled_capabilities_capability := main_component.get(
+            Capability.CUSTOM_DISABLED_CAPABILITIES
+        )
+    ) is not None:
```

Comment:
Would be nicer to reduce indentation of the next lines by changing this to:
```suggestion
    if (
        disabled_capabilities_capability := main_component.get(
            Capability.CUSTOM_DISABLED_CAPABILITIES
        )
    ) is None:
        return
```